Classes/Domain/Model/File.php

```diff
@@ -368,4 +368,23 @@ public function copy(File $fileToCopy) {
         return $this;
     }
 
+    /**
+     * Gets the full url of the file.
+     *
+     * @return string
+     */
+    public function getUrl()
+    {
+        $url = $this->getLink();
+
+        if (strpos(strtolower($url), "datastreams") === false) {
```

Comment:
Checking for "datastreams" in the URL is Fedora 3 specific, right? This will be a problem later, if we need to find all Fedora 3 related code. Could this be encapsulated into some Fedora 3 class? Or at least mark the spot with a FIXME-comment...

Reply:
For now, I would suggest solving it with a FIXME comment, I've already added one.